index.js

```diff
@@ -143,6 +145,24 @@ function RingPop(options) {
 
     this.destroyed = false;
     this.joiner = null;
+
+    this.startTime = Date.now(); //used for calculating uptime
+
+    this.tchannelVersion = getTchannelVersion(this);
+    this.ringpopVersion = packageJSON.version;
+}
+
+function getTchannelVersion(self)
+{
+    var tchannelPackage = require(path.join(path.dirname(require.resolve('tchannel')), 'package.json'));
```

Comment:
The problem here is that if we `require()` an invalid path/module, it'll throw. You'll need to add exception handling.